### DISCUSSION TITLE
[ANNOUNCEMENT] ClientCertificateRequested and ServerCertificateErrorDetected API changes are coming. This is a potentially breaking change.

### DISCUSSION THREAD

**Discussion by monica-ch** [MicrosoftEdge/WebView2Feedback#3957] [Announcements]
Currently in WebView2, we expose certificate metadata using [ICoreWebView2ClientCertificate](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2clientcertificate?view=webview2-1.0.1245.22)  and [ICoreWebView2Certificate](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2certificate?view=webview2-1.0.1245.22) in  [ClientCertificateRequested ](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2_5?view=webview2-1.0.1245.2) and [ServerCertificateErrorDetected](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2_14?view=webview2-1.0.1245.22#add_servercertificateerrordetected) events respectively. 

We noticed that [DEREncodedSerialNumber](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2clientcertificate?view=webview2-1.0.1245.22#get_derencodedserialnumber) isn't converted properly to a normal string and thereby want to display the serial number as a Base64 encoding of DEREncodedSerialNumber and the type remains same.

If your application is using the serial number, it could result in a breaking change and requires the application to update the code accordingly. 

Let us know if you have any questions or concerns.

**Comment by DexterWoo** [comment]
@monica-ch,  Is it possible to provide  ServerCertificateValidationCallback? which will allow people to validate the certificate even if it's trusted by the system?

**Comment by wilva** [comment]
That is something I had also seen and it would be great to see this addressed.

Is this issue perhaps related?
https://github.com/MicrosoftEdge/WebView2Feedback/issues/2346

**Comment by darbid** [comment]
I hope you might be able to help me understand what a [DEREncodedSerialNumber](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2clientcertificate?view=webview2-1.0.1245.22#get_derencodedserialnumber) is and why you are using it. Why not use the same property as the X509Certificate2 [SerialNumber](https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate2.serialnumber?view=net-6.0)

And could you by chance have an example C# code of converting the DerEncode?
